src/ol/source/Vector.js

```diff
@@ -1116,22 +1116,36 @@ class VectorSource extends Source {
    */
   removeFeatureInternal(feature) {
     const featureKey = getUid(feature);
+    if (!this.uidIndex_.hasOwnProperty(featureKey)) {
```

Comment:
Elsewhere we do this like
```suggestion
    if (!(featureKey in this.uidIndex_)) {
```